acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/BesuNode.java

```diff
@@ -300,6 +303,26 @@ private String getDiscoveryPort() {
     }
   }
 
+  public int selectP2pListenPort(final boolean staticP2pPort) {
+    return staticP2pPort ? getRandomPort() : 0;
+  }
+
+  int getP2pListenPort() {
+    return p2pListenPort;
+  }
+
+  private int getRandomPort() {
+    int port = 0;
+    try {
+      ServerSocket socket = new ServerSocket(0);
+      port = socket.getLocalPort();
+      socket.close();
+    } catch (final Exception e) {
```

Comment:
are we introducing a race condition here?

Reply:
Yes, this will 100% guaranteed cause intermittent failures. It's a bad anti-pattern to think you can find a free port this way because the most likely port that will next be used by the system is the one you just got, so it is very often in use before you actually start the real server.